src/app/business/profile/profile.component.ts

```diff
@@ -68,12 +72,51 @@ export class ProfileComponent implements OnInit {
     }
     getProfiles() {
         this.ProfileService.getProfiles().subscribe((res) => {
-            this.profiles = res.json();
-            this.profiles.forEach(item=>{
+            let str = res.json();
+            this.allProfileNameForCheck = [];
+            str.forEach(item=>{
                 if(item.storageType && item.storageType == "file"){
                     delete item.replicationProperties;
                 }
+                this.allProfileNameForCheck.push(item.name);
             })
+            this.getPools(str);
         });
     }
+    getPools(str) {
+        this.PoolService.getPools().subscribe((res) => {
```

Comment:
ideally, the backend must support 2 API

1. get StorageTypes(void), returns enum with all storage types
2. getPoolsForType(type StorageTypeEnumValue), get the pools only for that type
like, this all over the place, 

1. there will be checks, if-else blocks for file, block types etc.
2. when a new type gets added, multiple places code will need to change

Reply:
This requires the APIs to be redesigned and/or modified. We can discuss this.  I feel it is beyond the scope of this PR.

Reply:
LG...AB